"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const http_exception_1 = require("./http.exception");
describe('http.exception', () => {
    let httpError;
    beforeEach(() => {
        const request = {
            url: 'url',
        };
        const headers = {
            contentType: 'appli/json',
            keepAlive: true,
        };
        const data = {
            content: {
                key1: 'val1',
                key2: 'val2',
            },
        };
        httpError = {
            code: 'Error',
            message: 'Error message',
            request: request,
            response: {
                status: 500,
                statusText: 'Internal server Error',
                headers: headers,
                data: data,
            },
        };
    });
    describe('HttpError interface', () => {
        it('should have 4 properties', () => {
            expect(Object.keys(httpError).length).toEqual(4);
        });
    });
    describe('HttpException', () => {
        it('should be defined', () => {
            expect(http_exception_1.HttpException).toBeDefined();
        });
        it('should have an error property', () => {
            const httpException = new http_exception_1.HttpException(httpError);
            expect(httpException).toHaveProperty('error', httpError);
        });
    });
});
//# sourceMappingURL=http.exception.spec.js.map